src/group_theory/schreier.lean

```diff
@@ -178,4 +180,33 @@ begin
   simpa only [monoid_hom.mem_ker, subtype.ext_iff] using this,
 end
 
+/-- A bound for the size of the commutator subgroup in terms of the number of commutators. -/
+def card_commutator_bound (n : ℕ) := (n ^ (2 * n)) ^ (n ^ (2 * n + 1) + 1)
+
+/-- A theorem of Schur: A theorem of Schur: The size of the commutator subgroup is bounded in terms
+  of the number of commutators. -/
+lemma card_commutator_le_of_finite_commutator_set [finite (commutator_set G)] :
+  nat.card (commutator G) ≤ card_commutator_bound (nat.card (commutator_set G)) :=
+begin
+  have h1 := index_center_le_pow (closure_commutator_representatives G),
+  have h2 := card_commutator_dvd_index_center_pow (closure_commutator_representatives G),
+  rw card_commutator_set_closure_commutator_representatives at h1 h2,
+  rw card_commutator_closure_commutator_representatives at h2,
+  replace h1 := h1.trans (nat.pow_le_pow_of_le_right finite.card_pos
+    (rank_closure_commutator_representations_le G)),
+  replace h2 := h2.trans (pow_dvd_pow _ (add_le_add_right (mul_le_mul_right' h1 _) 1)),
+  rw ← pow_succ' at h2,
+  refine (nat.le_of_dvd _ h2).trans (nat.pow_le_pow_of_le_left h1 _),
+  exact pow_pos (nat.pos_of_ne_zero finite_index.finite_index) _,
+end
+
+/-- A theorem of Schur: A group with finitely many commutators has finite commutator subgroup. -/
+instance [finite (commutator_set G)] : finite (commutator G) :=
+begin
+  have h2 := card_commutator_dvd_index_center_pow (closure_commutator_representatives G),
+  refine nat.finite_of_card_ne_zero (λ h, _),
+  rw [card_commutator_closure_commutator_representatives, h, zero_dvd_iff] at h2,
+  exact finite_index.finite_index (pow_eq_zero h2),
+end
```

Comment:
I wonder why this is not proved using the lemma above?
Anyways LGTM.
maintainer merge

Reply:
It doesn't follow from the previous lemma since `nat.card` equals zero when the type is infinite.